Packages/com.unity.inputsystem/InputSystem/Devices/InputDeviceMatcher.cs

```diff
@@ -142,6 +142,17 @@ public InputDeviceMatcher WithManufacturer(string pattern, bool supportRegex = t
             return With(kManufacturerKey, pattern, supportRegex);
         }
 
+        /// <summary>
+        /// Add a pattern (simple string) to <see cref="patterns"/> to match a <see cref="InputDeviceDescription.manufacturer"/>.
+        /// </summary>
+        /// <param name="noRegExPattern">String to match - simple keyword search in a device manufacturer string (eg "SONY").</param>
+        /// <returns>The modified device matcher with the added pattern.</returns>
+        /// <seealso cref="InputDeviceDescription.manufacturer"/>
+        public InputDeviceMatcher WithManufacturerContains(string noRegExPattern)
+        {
+            return With(kManufacturerContainsKey, noRegExPattern, supportRegex:false);
```

Comment:
Given that supportRegex:false is passed, do you reallly need kManufacturerContainsKey.
It seems you could just use kManufacturerKey since the object type will remain a string

(but I may have missed something)

Reply:
The different key is needed to take the different path through MatchPercentage().

kManufacturerKey -> MatchSingleProperty -> string.Compare(...) == 0  // string/regex needs exactly match
kManufacturerContainsKey -> MatchSinglePropertyContains -> str.Contains(...) // string just needs to be in there somewhere